src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainer.java

```diff
@@ -135,6 +138,19 @@ public void onPutStackInSlot(int slotId, ItemStack itemstack, CallbackInfo ci) {
         }
     }
 
+    @Inject(method = "slotClick", at = @At(value = "HEAD"), cancellable = true)
+    public void onSlotClick(int slotId, int clickedButton, int mode, EntityPlayer player, CallbackInfoReturnable<net.minecraft.item.ItemStack> ci) {
+        if (slotId >= inventorySlots.size()) {
+            // Kick the player who's sending invalid packets and cancel the click
+            if (player instanceof EntityPlayerMP) {
+                ((EntityPlayerMP) player).playerNetServerHandler
+                        .kickPlayerFromServer("You have been kicked for attempting to click an out of bounds slot.");
+                ci.setReturnValue(null);
+                ci.cancel();
```

Comment:
`setReturnValue` cancels the callback automatically, so this method call is redundant